packages/connection-form/src/components/connection-form.tsx

```diff
@@ -451,7 +486,8 @@ function ConnectionForm({
         <div className={formHeaderStyles}>
           <div className={formHeaderContentStyles}>
             <H3 className={headingWithHiddenButtonStyles}>
-              {initialConnectionInfo.favorite?.name ?? 'New Connection'}
+              {(initialConnectionInfo.favorite?.name ?? 'New Connection') ||
+                'Edit Connection'}
```

Comment:
Drive-by to fix the case where you edit a connection that has no favorite name.

<img width="988" alt="Screenshot 2024-10-08 at 14 14 53" src="https://github.com/user-attachments/assets/5a8de135-a6ea-4ec9-9032-b02b58459d56">